core/invoker/src/main/scala/whisk/core/container/ContainerUtils.scala

```diff
@@ -164,14 +164,19 @@ trait ContainerUtils {
             val buffer = java.nio.ByteBuffer.allocate(remain)
             while (remain > 0) {
                 val read = channel.read(buffer)
-                if (read > 0)
-                    remain = read - read.toInt
+                if (read >= 0) {
+                    remain -= read
+                } else {
+                    // read < 0 means premature end of file => terminate loop
+                    remain = 0
+                }
```

Comment:
I don't know what happened to my comment. So here goes again: This log reader runs before tearing down a container to save the logs from an action container to a file for post inspection. Should we just nuke this code? There are also no tests for this code in general. This is not the log reader we use for activation log extraction.